src/Psy/Plugin/Manager.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+namespace Psy\Plugin;
+
+class Manager
+{
+    /** @var AbstractPlugin[] */
+    protected static $plugins = array();
+
+    /**
+     * @param AbstractPlugin $plugin
+     * @param $name
+     */
+    public static function register(AbstractPlugin $plugin, $name)
+    {
+        self::$plugins[$name] = $plugin;
+    }
```

Comment:
With the method you used, it is accessible though... it should also check that the plugin is not already registered